src/main/java/ua/com/javarush/oleksandr/reddit/redditcloneabstract/controller/PostController.java

```diff
@@ -30,55 +28,53 @@ public class PostController {
 
     private final PostService postService;
     private final PostMapper postMapper;
+    private final PostRequestValidator postRequestValidator;
+    private final BindingResultService bindingResultService;
 
     @PostMapping
-    public ResponseEntity<?> createPost(@RequestBody @Valid PostRequestDTO postRequestDTO,
+    public ResponseEntity<?> createPost(@RequestBody @Valid PostRequest postRequest,
                                         BindingResult bindingResult) {
 
-        if (bindingResult.hasErrors()) {
-            return handleErrorsFromBindingResult(bindingResult);
-        }
+        postRequestValidator.validate(postRequest, bindingResult);
+        bindingResultService.handle(bindingResult, PostCreationException::new);
 
-        var post = postMapper.dtoToPost(postRequestDTO);
+        var post = postMapper.dtoToPost(postRequest);
         postService.save(post);
 
         URI location = buildPostLocationUri(post.getId());
 
-        return ResponseEntity.created(location).build();
+        return ResponseEntity.status(HttpStatus.CREATED).body(location);
     }
 
     @GetMapping
-    public ResponseEntity<List<PostResponseDTO>> getAllPosts() {
+    public ResponseEntity<?> getAllPosts() {
 
-        List<PostResponseDTO> postResponseDtos = postService.findAll().stream()
+        List<PostResponse> postResponses = postService.findAll().stream()
                 .map(postMapper::postToDTO)
                 .toList();
 
-        return ResponseEntity.ok(postResponseDtos);
+        return ResponseEntity.status(HttpStatus.OK).body(postResponses);
     }
 
     @GetMapping("/{id}")
-    public ResponseEntity<PostResponseDTO> getPost(@PathVariable Long id) {
+    public ResponseEntity<?> getPost(@PathVariable Long id) {
 
         Optional<Post> foundPost = postService.findById(id);
 
         var foundPostDto = foundPost.map(postMapper::postToDTO)
-                .orElseThrow(() -> new ResponseStatusException(
-                        HttpStatus.NOT_FOUND,
-                        String.format("Post with id = %d not found", id)
-                ));
+                .orElseThrow(PostNotFoundException::new);
 
-        return ResponseEntity.ok(foundPostDto);
+        return ResponseEntity.status(HttpStatus.OK).body(foundPostDto);
     }
 
     @GetMapping("/by-user/{username}")
-    public ResponseEntity<List<PostResponseDTO>> getPostsByUsername(@PathVariable String username) {
+    public ResponseEntity<?> getPostsByUsername(@PathVariable String username) {
 
         var posts = postService.findPostsByUsername(username).stream()
                 .map(postMapper::postToDTO)
                 .toList();
 
-        return ResponseEntity.ok(posts);
+        return ResponseEntity.status(HttpStatus.OK).body(posts);
```

Comment:
why you don't like ok() method?

Reply:
It's very simple, I like to manually set the configuration